lms/djangoapps/grades/models.py

```diff
@@ -690,6 +676,59 @@ def get_override(cls, user_id, usage_key):
         except PersistentSubsectionGradeOverride.DoesNotExist:
             pass
 
+    @classmethod
+    def update_or_create_override(
+        cls, requesting_user, subsection_grade_model, feature=None, action=None, **override_data
+    ):
+        """
+        Creates or updates an override object for the given PersistentSubsectionGrade.
+        Args:
+            requesting_user: The user that is creating the override (so we can record this action in
+            a PersistentSubsectionGradeOverrideHistory record).
+            subsection_grade_model: The PersistentSubsectionGrade object associated with this override.
+            override_data: The parameters of score values used to create the override record.
+        """
+        override, _ = PersistentSubsectionGradeOverride.objects.update_or_create(
+            grade=subsection_grade_model,
+            defaults=cls._prepare_override_params(subsection_grade_model, override_data),
+        )
+
+        feature = feature or PersistentSubsectionGradeOverrideHistory.GRADEBOOK
```

Comment:
hmm.. why couple the gradebook feature, even by name, into core grades?

Reply:
Good question, I'm pretty sure I can un-couple it here (although it will still be coupled in the `OVERRIDE_FEATURES` field on L739 below).